### PR TITLE
Avoid caching empty model listings

### PR DESCRIPTION
**Pull request by FrancescoMonti-source**
## Summary
- Prevent caching when backend returns no models and clear any existing entry
- Ensure local model fetches return zero rows without status when empty
- Add regression test for refresh with empty model list

## Testing
- `R -q -e 'devtools::test()'` *(fails: command not found)*
- `apt-get update` *(fails: repository '... InRelease' is not signed)*

------
https://chatgpt.com/codex/tasks/task_e_68b8372af6888321be1d307a5164548e

### CODE REVIEW

File: R/models_cache.R [hunk 1]
```diff
@@ -395,14 +399,21 @@
...
+      if (length(mods) > 0) {
+        .cache_put(provider, base_url, mods)
+        tmp <- .cache_get(provider, base_url)
+        ts <- if (!is.null(tmp)) tmp$ts else ts
+      } else {
+        ts <- as.numeric(Sys.time())
+      }
     }
-    .row_df(provider, base_url, mods, "installed", src, ts, status = live$status)
+    status <- if (nrow(mods) > 0) live$status else NA_character_
```
[the remark this answers is not in the document]
Comment (FrancescoMonti-source) [reply R/models_cache.R:415]: @codex agree
